assets/app/scripts/extensions/javaLink.js

```diff
@@ -1,86 +1,79 @@
+              'ng-href="item.url">',
+              'Open Java Console',
+            '</a>',
+          '</div>',
+        '</div>'
+      ].join('');
 
-    HawtioExtension.add('container-links', function ($scope) {
-      var container = $scope.container;
-      if (!container) {
-        return;
-      }
-      var jolokiaPort = _.find((container.ports || []), function(port) {
-        return port.name && port.name.toLowerCase() === 'jolokia';
-      });
-      if (!jolokiaPort) {
-        return;
-      }
-      var pod = $scope.$eval('podTemplate');
-      // TODO distinguish between pod and pod templates for now...
-      if (!pod || !pod.status || pod.status.phase !== 'Running') {
-        return;
-      }
-      var containerStatuses = pod.status.containerStatuses;
-      var containerStatus = _.find(containerStatuses, function(status) {
-        return status.name === container.name;
-      });
-      if (!containerStatus || !containerStatus.ready) {
-        return;
-      }
-      var podName = pod.metadata.name;
-      var namespace = pod.metadata.namespace;
-      $scope.jolokiaUrl = ProxyPod(namespace, podName, jolokiaPort.containerPort).segment('jolokia/').toString();
-      $scope.gotoContainerView = function($event, container, jolokiaUrl) {
-        $event.preventDefault();
-        $event.stopPropagation();
-        var returnTo = window.location.href;
-        var title = container.name || 'Untitled Container';
-        var token = AuthService.UserStore().getToken() || '';
-        var targetURI = new URI().path(BaseHref)
-                                 .segment('java/') // Must have a trailing slash to avoid runtime errors in Safari
-                                 .hash(token)
-                                 .query({
-                                   jolokiaUrl: jolokiaUrl,
-                                   title: title,
-                                   returnTo: returnTo
-                                 });
-        window.location.href = targetURI.toString();
+      var makeJolokiaUrl = function(namespace, podName, port) {
+        return new URI(DataService.url({
+                    resource: 'pods/proxy',
+                    // always use https when connecting to jolokia in a pod
+                    name: ['https', podName, port || ''].join(':'),
+                    namespace: namespace
+                  }))
+                  .segment('jolokia/');
       };
-      var answer = $compile(template)($scope);
-      return answer;
-    });
-  });
 
-hawtioPluginLoader.addModule('openshiftConsoleExtensions');
+      extensionRegistry.add('container-links', _.spread(function(container, pod) {
+        var jolokiaPort = _.find((container.ports || []), function(port) {
+          return port.name && port.name.toLowerCase() === 'jolokia';
+        });
+        if (!jolokiaPort) {
+          return;
+        }
+        if(_.get(pod, 'status.phase') !== 'Running') {
```

Comment:
just to make sure everything's working smoothly, did you try going to a pod's page (or the overview) when the pod was still in pending, and then saw the connect link appear after the pod started running?